graphql_compiler/compiler/metadata.py

```diff
@@ -154,11 +161,20 @@ def get_location_info(self, location):
                                  u'{}'.format(location))
         return location_info
 
+    @property
+    def tags(self):
+        """Return a dict of tag name to TagInfo for the query."""
+        return self._tags
+
+    def record_tag_info(self, tag_name, tag_info):
+        """Record information about the tag."""
+        self._tags[tag_name] = tag_info
```

Comment:
Is it ever okay to overwrite an existing tag's info? If not, let's assert that the tag being recorded hasn't been recorded previously.